rokka/operations_gen.go

```diff
@@ -0,0 +1,154 @@
+	"text/template"
+	"time"
+
+	"github.com/rokka-io/rokka-go/rokka"
+)
+
+type operationProperty struct {
+	Name string
+	Type string
+}
+
+type operation struct {
+	Name       string
+	Properties []operationProperty
+	Required   []string
+	OneOf      []string
+}
+
+var typeMap = map[string]string{
+	"integer": "int",
+	"boolean": "bool",
+	"string":  "string",
+	"number":  "float64",
+}
+
+func toStringSlice(list []interface{}) []string {
+	slice := make([]string, len(list))
+	for i, v := range list {
+		slice[i] = v.(string)
+	}
+	return slice
+}
+
+func main() {
+	c := rokka.NewClient(&rokka.Config{})
+
+	res, err := c.GetOperations()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	f, err := os.Create("operations_objects.go")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	operations := make([]operation, 0)
+	for name, value := range res {
+		properties := make([]operationProperty, 0)
+		if propertiesMap, ok := value["properties"].(map[string]interface{}); ok {
+			for propName, propValue := range propertiesMap {
+				propValueMap := propValue.(map[string]interface{})
+				p := operationProperty{
+					propName,
+					typeMap[propValueMap["type"].(string)],
+				}
+				properties = append(properties, p)
+			}
+		}
+		var required, oneOf []string
+		if list, ok := value["required"]; ok {
+			required = toStringSlice(list.([]interface{}))
+		}
+		if list, ok := value["oneOf"]; ok {
+			oneOf = toStringSlice(list.([]interface{}))
+		}
+		o := operation{
+			name,
+			properties,
+			required,
+			oneOf,
+		}
+		operations = append(operations, o)
+	}
+
+	err = packageTemplate.Execute(f, struct {
+		Timestamp  time.Time
+		Operations []operation
+	}{
+		Timestamp:  time.Now(),
+		Operations: operations,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+var funcMap = template.FuncMap{
+	"title": strings.Title,
+}
+
+var packageTemplate = template.Must(template.New("").Funcs(funcMap).Parse(`// Code generated by go generate; DO NOT EDIT.
+// This file was generated at {{ .Timestamp }}
+package rokka
+
+import (
+	"errors"
+	"fmt"
+)
+
+{{- range .Operations }}
+	// {{ title .Name }}Operation is an auto-generated operation object specified by the rokka API.
+	{{- if (or .OneOf .Required) }}
+	// Missing required properties are treated as errors.
+	{{- end }}
+	//
+	// See: https://rokka.io/documentation/references/operations.html
+	func {{ title .Name }}Operation(options map[string]interface{}) (map[string]interface{}, error) {
+		{{- range .Required }}
+			if _ ,ok := options["{{ . }}"]; !ok {
+				return nil, errors.New("option \"{{ . }}\" is required")
+			}
+		{{- end}}
+		{{- if .OneOf }}
+		valid := false
+		{{- range .OneOf }}
+			if _ ,ok := options["{{ . }}"]; ok {
+				valid = true
+			}
+		{{- end}}
+		if !valid {
+			return nil, errors.New("one of \"{{ .OneOf }}\" is required")
+		}
+		{{- end}}
+		{{- if .Properties }}
+		allowed := map[string]interface{}{
+			{{ range .Properties -}}
+				"{{ .Name }}": {{ .Type }}.(type),
```

Comment:
that doesn't work (as well as below the if for the type). To figure out the type you need to use reflection.